x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -0,0 +1,330 @@
+		if len(output.Values) == 0 {
+			continue
+		}
+		switch *output.Id {
+		case "cpu1":
+			event.MetricSetFields.Put("cpu.total.pct", output.Values[0])
+		case "cpu2":
+			event.MetricSetFields.Put("cpu.credit_usage", output.Values[0])
+		case "cpu3":
+			event.MetricSetFields.Put("cpu.credit_balance", output.Values[0])
+		case "cpu4":
+			event.MetricSetFields.Put("cpu.surplus_credit_balance", output.Values[0])
+		case "cpu5":
+			event.MetricSetFields.Put("cpu.surplus_credits_charged", output.Values[0])
+		case "network1":
+			event.MetricSetFields.Put("network.in.packets", output.Values[0])
+		case "network2":
+			event.MetricSetFields.Put("network.out.packets", output.Values[0])
+		case "network3":
+			event.MetricSetFields.Put("network.in.bytes", output.Values[0])
+		case "network4":
+			event.MetricSetFields.Put("network.out.bytes", output.Values[0])
+		case "disk1":
+			event.MetricSetFields.Put("diskio.read.bytes", output.Values[0])
+		case "disk2":
+			event.MetricSetFields.Put("diskio.write.bytes", output.Values[0])
+		case "disk3":
+			event.MetricSetFields.Put("diskio.read.ops", output.Values[0])
+		case "disk4":
+			event.MetricSetFields.Put("diskio.write.ops", output.Values[0])
+		case "status1":
+			event.MetricSetFields.Put("status.check_failed", output.Values[0])
+		case "status2":
+			event.MetricSetFields.Put("status.check_failed_system", output.Values[0])
+		case "status3":
+			event.MetricSetFields.Put("status.check_failed_instance", output.Values[0])
+		}
+	}
+	report.Event(event)
+}
+
+func getInstancesPerRegion(svc ec2iface.EC2API) (instanceIDs []string, instancesOutputs map[string]ec2.Instance, err error) {
+	instancesOutputs = map[string]ec2.Instance{}
+	output := ec2.DescribeInstancesOutput{NextToken: nil}
+	init := true
+	for init || output.NextToken != nil {
+		init = false
+		describeInstanceInput := &ec2.DescribeInstancesInput{
+			Filters: []ec2.Filter{
+				{
+					Name:   awssdk.String("instance-state-name"),
+					Values: []string{"running"},
+				},
+			},
+		}
+
+		req := svc.DescribeInstancesRequest(describeInstanceInput)
+		output, err := req.Send()
+		if err != nil {
+			fmt.Println("Error DescribeInstances: ", err)
+			return nil, nil, err
+		}
+
+		for _, reservation := range output.Reservations {
+			for _, instance := range reservation.Instances {
+				instanceID := *instance.InstanceId
+				instanceIDs = append(instanceIDs, instanceID)
+				instancesOutputs[instanceID] = instance
+			}
+		}
+	}
+	return
+}
+
+func getMetricDataPerRegion(instanceID string, nextToken *string, svc cloudwatchiface.CloudWatchAPI) (*cloudwatch.GetMetricDataOutput, error) {
+	//TODO:remove hard coded variables
+	endTime := time.Now()
+	duration, err := time.ParseDuration("-10m")
+	if err != nil {
+		fmt.Println("Error ParseDuration: ", err)
+		return nil, err
+	}
+
+	startTime := endTime.Add(duration)
+
+	//TODO:add function getMetricNames from environment variables
```

Comment:
what would this `getMetricNames` do?

Reply:
Haha I thought I saw a getMetricName method somewhere in aws api to query all the available metric names from cloudwatch ec2. But I guess it's not necessary here and probably will make the code even messier.